Issue #4722 - remove websocket-servlet

lachlan-roberts opened the pull request:
**Issue #4722**

Removed the `websocket-servlet` layer which was resulting in exposing `websocket-core` classes.

- `WebSocketServlet` and `WebSocketServletFactory` have been merged into `JettyWebSocketServlet` and `JettyWebSocketServletFactory` which were extending them previously.

- The other classes including `WebSocketUpgradeFilter` and `WebSocketMapping` were moved into the new package `org.eclipse.jetty.websocket.util.server` of `websocket-util`, but maybe these should be in `websocket-core` instead

lachlan-roberts committed:
Issue #4722 - remove websocket-servlet

Signed-off-by: Lachlan Roberts <lachlan@webtide.com>

lachlan-roberts committed:
Issue #4722 - fix typo in JavaxWebSocketConfiguration

Signed-off-by: Lachlan Roberts <lachlan@webtide.com>

lachlan-roberts commented:
I'm getting a failure on this branch for `DistributionTests.testSimpleWebAppWithWebsocket` saying 
```
java.util.ServiceConfigurationError: org.slf4j.spi.SLF4JServiceProvider: Provider org.eclipse.jetty.logging.JettyLoggingServiceProvider not found
```
Any idea what would cause this @joakime  @olamy , is this something to do with the recent logging changes?

olamy committed:
those jars are provided as they come from jetty distribution if not they try to load slf4j impl

Signed-off-by: olivier lamy <oliver.lamy@gmail.com>

olamy commented:
@lachlan-roberts should be fixed with this commit https://github.com/eclipse/jetty.project/commit/5199e78587276962028d39b5c5ddfa16a963b8f9

joakime reviewed (approved):
:+1:

lachlan-roberts committed:
Issue #4722 - changes from review

- Fix packages exposed in the websocket configuration
- Make servlet dependency for websocket-util optional

Signed-off-by: Lachlan Roberts <lachlan@webtide.com>

joakime reviewed (approved):
Still a :+1:  from me

lachlan-roberts committed:
Issue #4722 - fix build, websocket util exporting jetty servlet with jpms

Signed-off-by: Lachlan Roberts <lachlan@webtide.com>

lachlan-roberts commented:
@joakime the change to make `jetty-servlet` an optional dependency for `websocket-util` has caused some problems.

The last commit got it working working but giving a JPMS warning as there are servlet classes in the public signature of `websocket-util` server package.
```
Warning:(106, 19) java: class org.eclipse.jetty.servlet.FilterHolder in module org.eclipse.jetty.servlet is not indirectly exported using requires transitive
```

As an alternative I could introduce another module which is shared only between both `websocket-jetty-server` and `websocket-javax-server`, called something like `websocket-server-common` and then move the `WebSocketUpgradeFilter` and other classes here. I still think `WebSocketServlet` does not belong in that package and the current changes to merge that with `JettyWebSocketServlet` is a good change.

gregw commented:
@lachlan-roberts I'm loath to introduce more modules.  I'm sure JPMS can deal with optional/provided dependencies.... ask @sbordet for advice or look for other examples of provided dependencies in the code base.

lachlan-roberts commented:
@gregw I asked @sbordet for advice and he said it would be best to introduce a new `websocket-util-server` module.

lachlan-roberts committed:
Issue #4722 - create new module websocket-util-server

- This module contains the WebSocketUpgradeFilter.
- Also has an internal package with the components used
  to implement websocket upgrades common to both the
  jetty and javax websocket implementations.

Signed-off-by: Lachlan Roberts <lachlan@webtide.com>

lachlan-roberts commented:
I pushed a commit to introduce a `websocket-util-server` module which has the `WebSocketUpgradeFilter` and other classes in internal package.

It only exports the internal package to `websocket-jetty-server` and `websocket-javax-server` and exports the base package (containing only the upgrade filter) to all.

lachlan-roberts committed:
Issue #4722 - fix typo for websocket-util-server jar

Signed-off-by: Lachlan Roberts <lachlan@webtide.com>

joakime commented:
The osgi tests on this branch are failing.

lachlan-roberts committed:
Merge remote-tracking branch 'origin/jetty-10.0.x' into jetty-10.0.x-WebSocketServlet

lachlan-roberts commented on jetty-osgi/pom.xml:
```diff
@@ -90,7 +90,7 @@
       <dependency>
         <groupId>org.eclipse.jetty</groupId>
-        <artifactId>jetty-websocket</artifactId>
+        <artifactId>jetty-websocket</artifactId> <!-- TODO: is this even right? -->
```
@janbartel while debugging I saw this, any idea what this is for? I don't know what this is referring to, I don't think there is any `jetty-websocket` artifactId in jetty 10 so I thought this should be an errror.

lachlan-roberts committed:
Fix broken OSGI tests due to websocket-util-server

Signed-off-by: Lachlan Roberts <lachlan@webtide.com>

lachlan-roberts committed:
Fix broken WebSocket UTF-8 continuation test

Signed-off-by: Lachlan Roberts <lachlan@webtide.com>

lachlan-roberts committed:
Merge remote-tracking branch 'origin/jetty-10.0.x' into jetty-10.0.x-WebSocketServlet

Signed-off-by: Lachlan Roberts <lachlan@webtide.com>

lachlan-roberts commented:
@gregw || @joakime can I get a tick after the new changes where I added the module `jetty-util-server`? I'd like to get this merged before we do another release of 10.

gregw reviewed (approved):
LGTM.
